src/main/java/cubicchunks/server/CubeIO.java

```diff
@@ -542,7 +542,7 @@ private static NBTTagCompound writeCubeToNbt(final Cube cube) {
 		// tile entities
 		NBTTagList nbtTileEntities = new NBTTagList();
 		nbt.setTag("TileEntities", nbtTileEntities);
-		for (TileEntity blockEntity : cube.getTileEntityMap().values()) {
+		for (TileEntity blockEntity : cube.getTileEntityMap()) {
```

Comment:
This is just wrong. cube.getTileEntityMap() returns a Map, which isn't Iterable.